### PR TITLE
Correctly identify failures in problem daemon starting.

### PR DESCRIPTION
**Pull request by xueweiz**
This is for #298 

/assign @wangzhen127 
/cc @andyxning @kewu1992

### CODE REVIEW

**Comment by wangzhen127** [comment]
/ok-to-test
Can you also update the comments of the `Monitor` interface's, especially the `Start()` function (about channel can be nil)?

**Comment by xueweiz** [comment]
Thanks for the review @wangzhen127 !
I just updated the comment near the `Monitor` interface.

**Comment by xueweiz** [comment]
The test failure at https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/node-problem-detector/299/pull-npd-e2e-kubernetes-gce-gci-custom-flags/1143599705079418880 seems to be an infra problem, and a node is not being created successfully.

```
Detected 3 ready nodes, found 3 nodes out of expected 4. Your cluster may not be fully functional.
NAME                                     STATUS                     ROLES    AGE   VERSION
e2e-9e739a75cb-3dfbc-master              Ready,SchedulingDisabled   <none>   25m   v1.16.0-alpha.0.1498+68aaf8b91fcae9
e2e-9e739a75cb-3dfbc-minion-group-4z1s   Ready                      <none>   25m   v1.16.0-alpha.0.1498+68aaf8b91fcae9
e2e-9e739a75cb-3dfbc-minion-group-grf5   Ready                      <none>   25m   v1.16.0-alpha.0.1498+68aaf8b91fcae9
2019/06/25 19:51:44 process.go:155: Step './hack/e2e-internal/e2e-up.sh' finished in 29m31.882406574s
```

File: pkg/types/types.go [hunk 1]
```diff
@@ -103,6 +103,8 @@ const (
 // the rules.
```
Comment (wangzhen127) [inline pkg/types/types.go]: Can you also update the comments for the monitor inteface? It is not just for condition and event now.

Comment (xueweiz) [reply pkg/types/types.go]: Good catch! Done.

**Pull request merged**